TR001 -  Header Page

muruganvenkatesan opened the pull request:
created header page to show logo,project name,hamburger icon and current logged in user name  for thervu app

muruganvenkatesan committed:
TR001 -  Home page creation

Keerthu94 commented on package.json:
```diff
@@ -3,6 +3,8 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@material-ui/core": "^4.9.10",
```
Why added material-ui when we decided not to use? Is this also needed for icons?

muruganvenkatesan replied:
Yes, material ui core is needed for icons

Keerthu94 replied:
ok

Keerthu94 commented on src/header/header.component.js:
```diff
@@ -0,0 +1,36 @@
+
+const header = props =>{
+  
+  let projectNameClassName = ["project-name-content"];
```
Why didnt you use const?

muruganvenkatesan replied:
I had a thought of changing class name so i used let, i will use const here i'm pushing values to assigned object(array) not reassign object(array).

Keerthu94 commented on src/header/header.component.js:
```diff
@@ -0,0 +1,36 @@
+import Assignment from '@material-ui/icons/Assignment';
+import Hamburger from '@material-ui/icons/Menu';
+
+const header = props =>{
```
space after arrow

muruganvenkatesan replied:
okay I'll correct

Keerthu94 commented on src/header/header.component.js:
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import './header.css';
```
Why dont you use css modules?

muruganvenkatesan replied:
When i run eject command i got only two webpack config files in my config folder , As per the max and others instructions have to changes for modules in webpack.config.dev.js and webpack.config.prod.js but i got only one js file (webpack.config.js) so i didn't  change modules and use.

Keerthu94 commented on src/header/header.component.js:
```diff
@@ -0,0 +1,36 @@
+        <div className="logo-content">
+          <Assignment style={{ fontSize: "190%" }} className="thervu-logo" /> 
+        </div> 
+        <div className={projectNameClassName.join(" ")}>
```
Use single quotes when using js

muruganvenkatesan replied:
okay, I will change

muruganvenkatesan committed:
TR001 - Header Page changes done based on review comments
1)Used css Modules
2)Destructured props and used in header
3)Used template literals instead of array
4)Used single quotes istead of double quotes
5)Changed  App as container then added handler,state.

Keerthu94 commented on src/App.js:
```diff
@@ -1,26 +1,30 @@
+  
+  onChangeSideDrawer = () => {
+    this.setState(state => ({
+      isCloseSideDrawer : !state.isCloseSideDrawer
```
name it as prevState

muruganvenkatesan replied:
Ok I'll change

Keerthu94 commented on src/App.js:
```diff
@@ -1,26 +1,30 @@
+      <div className={classes['app']}>
+       <Header 
+         isCloseSideDrawer={this.state.isCloseSideDrawer} 
+         onChangeSideDrawer={this.onChangeSideDrawer} />
```
FInd out which is better, binding or using arrow functions

Keerthu94 replied:
change onChangeSideDrawer to onToggleSideDrawer

muruganvenkatesan committed:
TR001 - Changes based upon review comments and 2spaces  for header page

muruganvenkatesan committed:
TR001 - Header Page After resolved conflict.

muruganvenkatesan committed:
TR001 - Removed comment line
